### PR TITLE
Add flexible windowing syntax to yaml.

### PR DESCRIPTION
**Pull request by robertwb**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/yaml/readme_test.py [hunk 1]
```diff
@@ -85,13 +86,46 @@ def guess_name_and_type(expr):
     output_schema = [
         guess_name_and_type(expr) for expr in m.group(1).split(',')
     ]
-    return next(iter(inputs.values())).pipeline | beam.Create(
-        [beam.Row(**{name: typ()
-                     for name, typ in output_schema})])
+    output_element = beam.Row(**{name: typ() for name, typ in output_schema})
+    return next(iter(inputs.values())) | beam.Map(
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/readme_test.py:90]: Note: previously the output didn't derive from the input so no windowing was inherited.

**Comment by robertwb** [comment]
R: @twang126

File: sdks/python/apache_beam/yaml/README.md [hunk 2]
```diff
@@ -215,6 +215,157 @@ pipeline:
...
+    - type: SomeAggregation
+      windowing:
+        type: sliding
+        size: 60
+        period: 10
+    - type: WriteToPubSub
+      topic: anotherPubSubTopic
+```
+
+Note that the `Sql` operation itself often a from of aggregation, and applying
```
Comment (twang126) [inline sdks/python/apache_beam/yaml/README.md:262]: nit: maybe reword to `Note that the `Sql` operation itself  **is** often a from of aggregation, and applying a windowing ~which~ will cause all grouping to be done per window.`

Comment (robertwb) [reply sdks/python/apache_beam/yaml/README.md]: Thanks. Fixed.

File: sdks/python/apache_beam/yaml/README.md [hunk 3]
```diff
@@ -215,6 +215,157 @@ pipeline:
...
+
+```
+pipeline:
+  type: chain
+  transforms:
+    - type: ReadFromPubSub
+      topic: myPubSubTopic
+      windowing:
+        type: fixed
+        size: 60
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/yaml/README.md:316]: TBD. They'll probably be properties here. (I thought about requiring an extra level of nesting by putting this under `windowfn` but that felt needlessly verbose.

**Review by twang126 — APPROVED** [review]
LGTM!

File: sdks/python/apache_beam/yaml/yaml_transform_test.py [hunk 4]
```diff
@@ -166,6 +166,121 @@ def test_csv_to_json(self):
...
+            - type: CreateTimestamped
+              elements: [0, 1, 2, 3, 4, 5]
+            - type: SumGlobally
+          windowing:
+            type: fixed
+            size: 4
+          ''',
+          providers=TEST_PROVIDERS)
+      assert_that(result, equal_to([6, 9]))
+
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/yaml/yaml_transform_test.py:282]: Yes, the last test in the README has multiple windowings.

**Comment by robertwb** [comment]
The apache_beam.io.tfrecordio_test.TestEnd2EndWriteAndRead.test_end2end_auto_compression failure is unrelated.
